source/extensions/filters/network/thrift_proxy/router/router_impl.h

```diff
@@ -36,15 +36,34 @@ class RouteEntryImplBase : public RouteEntry,
   // Router::Route
   const RouteEntry* routeEntry() const override;
 
-  virtual RouteConstSharedPtr matches(const MessageMetadata& metadata) const PURE;
+  virtual RouteConstSharedPtr matches(const MessageMetadata& metadata,
+                                      uint64_t random_value) const PURE;
 
 protected:
-  RouteConstSharedPtr clusterEntry() const;
+  RouteConstSharedPtr clusterEntry(uint64_t random_value) const;
   bool headersMatch(const Http::HeaderMap& headers) const;
 
 private:
+  class WeightedClusterEntry : public RouteEntry, public Route {
+  public:
+    WeightedClusterEntry(
+        const envoy::config::filter::network::thrift_proxy::v2alpha1::WeightedCluster_ClusterWeight&
+            cluster);
+
+    uint64_t clusterWeight() const { return cluster_weight_; }
+    const std::string& clusterName() const override { return cluster_name_; }
```

Comment:
Envoy convention is to put a comment with the name of the interface being implemented before a group of methods. So `// RouteEntry` above clusterName and `// Route` above routeEntry.